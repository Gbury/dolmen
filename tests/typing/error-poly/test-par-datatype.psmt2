(set-logic ALL)
(set-info :smt-lib-version 2.6)

(declare-datatype list ( par (X) (
  ( nil )
  ( cons ( head X) ( tail ( list X ))))))

(define-fun singleton (par (a) ((x a)) (list a) (cons x nil)))

(assert (par (a) (forall ((x a)) (not (= (cons x nil) (cons 1 nil))))))

(check-sat)
