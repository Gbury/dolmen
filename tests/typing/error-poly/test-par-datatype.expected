File "tests/typing/error-poly/test-par-datatype.psmt2", line 10, character 38-67:
Error The term: cons(int; 1, nil(int)) has type list(int)
      but was expected to be of type list(a)
